### PR TITLE
Improve handling for http responses that are too slow and / or too large

### PR DESCRIPTION
**Pull request by shyamnamboodiripad**
Updates the `HttpRequestKernel` to send leading `DisplayEvent`s with partial information when the complete http response is too slow and / or too large. These leading events are then used to drive progress display to improve end-user perceived performance.

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive.HttpRequest/HttpRequestKernel.cs [hunk 1]
```diff
@@ -97,44 +94,135 @@ async Task IKernelCommandHandler<SubmitCode>.HandleAsync(SubmitCode command, Ker
...
+        }
 
-            context.Display(response);
+        void ClearDisplayedValue()
+        {
+            if (displayedValue is not null)
+            {
+                var htmlFormattedValue = new FormattedValue(HtmlFormatter.MimeType, "<span/>");
+                var plainTextFormattedValue = new FormattedValue(PlainTextFormatter.MimeType, string.Empty);
+                displayedValue.Update(null, htmlFormattedValue, plainTextFormattedValue);
```
Comment (shyamnamboodiripad) [inline src/Microsoft.DotNet.Interactive.HttpRequest/HttpRequestKernel.cs:187]: Avoid need to pass null.

Comment (shyamnamboodiripad) [reply src/Microsoft.DotNet.Interactive.HttpRequest/HttpRequestKernel.cs]: (Fixed)

File: src/Microsoft.DotNet.Interactive/FormattedValue.cs [hunk 2]
```diff
@@ -19,35 +19,39 @@ public FormattedValue(string mimeType, string value)
...
         }
 
-        return new FormattedValue(mimeType, value.ToDisplayString(mimeType));
+        return new FormattedValue(mimeType, value.ToDisplayString(mimeType), suppressDisplay);
     }
 
     public static IReadOnlyList<FormattedValue> CreateManyFromObject(object value, params string[] mimeTypes)
+        => CreateManyFromObject(value, suppressDisplay: false, mimeTypes);
+
+    public static IReadOnlyList<FormattedValue> CreateManyFromObject(object value, bool suppressDisplay = false, params string[] mimeTypes)
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive/FormattedValue.cs:44]: Once we start having this many properties on the type, I'm not sure the immutability is helping any, and adding the overload is a lot more surface area than for example just making `FormattedValue.SuppressDisplay` settable.

Comment (shyamnamboodiripad) [reply src/Microsoft.DotNet.Interactive/FormattedValue.cs]: Good call. Removed the overloads and made the property settable in the latest commit.

File: src/Microsoft.DotNet.Interactive.HttpRequest.Tests/HttpRequestKernelTests.cs [hunk 3]
```diff
@@ -363,12 +393,174 @@ public async Task produces_json_html_and_plain_text_formatted_values()
...
+    [Fact]
+    public async Task produces_empty_response_for_slow_responses()
+    {
+        HttpRequestMessage request = null;
+        var slowHandler = new InterceptingHttpMessageHandler(async (message, _) =>
+        {
+            request = message;
+            var response = new HttpResponseMessage(HttpStatusCode.OK);
+            response.RequestMessage = message;
+            await Task.Delay(1500);
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.HttpRequest.Tests/HttpRequestKernelTests.cs:422]: Is this delay a magic number of some kind that indicates a "slow" response? Putting a hard-coded delay like this into a test slows tests down while also likely being nondeterministic. Let's work on improving this test design.

File: src/Microsoft.DotNet.Interactive.HttpRequest/Formatting/HttpResponseFormattingExtensions.cs [hunk 4]
```diff
@@ -48,6 +75,43 @@ internal static void FormatAsHtml(this HttpResponse? response, FormatContext con
...
+        PocketView? output = null;
+
+        if (response is HttpResponse fullResponse)
+        {
+            output = fullResponse.FormatAsHtml();
+        }
+        else if (response is PartialHttpResponse partialResponse)
+        {
+            output =
+                PocketViewTags.div[@class: ContainerClass](
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.HttpRequest/Formatting/HttpResponseFormattingExtensions.cs:87]: These can be more concise if you add this at the top:

```csharp
using static Microsoft.DotNet.Interactive.Formatting.PocketViewTags;
```

File: src/Microsoft.DotNet.Interactive.HttpRequest.Tests/HttpRequestKernelTests.cs (hunk 3 above)
Comment (shyamnamboodiripad) [reply src/Microsoft.DotNet.Interactive.HttpRequest.Tests/HttpRequestKernelTests.cs:422]: The spinner is displayed on responses that take longer than a second. Yes glad to improve this if you have suggestions.

File: src/Microsoft.DotNet.Interactive.HttpRequest/Formatting/HttpResponseFormattingExtensions.cs (hunk 4 above)
Comment (shyamnamboodiripad) [reply src/Microsoft.DotNet.Interactive.HttpRequest/Formatting/HttpResponseFormattingExtensions.cs]: Fixed in latest iteration.

**Review by jonsequitur — APPROVED** [review]

**Pull request merged**